src/js/client/editprefset-genericSettingsTable-widget.js

```diff
@@ -0,0 +1,206 @@
+
+        settingsFilter: "mysettings",
+        settingsSearch: "",
+
+        commonTermUsageCounts: {}
+    },
+    templates: {
+        initial: "editprefset-genericSettingsTable-widget"
+    },
+    invokers: {
+        updateSettingsFilter: {
+            funcName: "gpii.devpmt.updateSettingsFilter",
+            args: ["{that}", "{that}.dom.commonTermRow", "{that}.model.settingsFilter", "{that}.model.settingsSearch"]
+        },
+        searchSettings: {
+            funcName: "gpii.devpmt.searchSettings",
+            args: ["{that}", "{that}.dom.settingsSearchInput"]
+        }
+    },
+    bindings: {
+    },
+    selectors: {
+        commonTermRow: ".pmt-commonterm-row",
+        prefsFilter: "#pmt-filter-container",
+        valueDisplayCell: ".pmt-value-display",
+
+        // Generic Prefs Filters
+        settingsSearchInput: "#pmt-settings-search-input",
+        addContextButton: ".pmt-add-context-button",
+
+        // mineAllWidget
+        mineAllSwitchContainer: "#pmt-mineAllSwitch-container",
+
+        // PSP items
+        pspShowCheckboxes: ".pmt-psp-show-checkbox",
+        pspMemoryCheckboxes: ".pmt-psp-memory-checkbox"
+    },
+    components: {
+        mineAllSwitch: {
+            type: "gpii.devpmt.mineAllSwitch.default",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.mineAllSwitchContainer",
+            options: {
+                selectors: {
+                    initial: "#pmt-mineAllSwitch-area"
+                },
+                model: {
+                    settingsFilter: "{genericSettingsTableWidget}.model.settingsFilter",
+                    allSettingsCount: "{genericSettingsTableWidget}.model.commonTermUsageCounts.all",
+                    mySettingsCount: "{genericSettingsTableWidget}.model.commonTermUsageCounts.npset"
+                }
+            }
+        },
+        filter: {
+            type: "gpii.devpmt.filterWidget",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.prefsFilter",
+            options: {
+                selectors: {
+                    initial: "#pmt-common-filter-area"
+                },
+                model: {
+                    filterText: "{genericSettingsTableWidget}.model.settingsSearch"
+                }
+            }
+        }
+    },
+    markupEventBindings: {
+        valueDisplayCell: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editValueEvent"
+        },
+        addContextButton: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.events.openAddContextDialog.fire"
+        },
+        pspShowCheckboxes: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editPspShow"
+        },
+        pspMemoryCheckboxes: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editPspMemory"
+        }
+    },
+    listeners: {
+        "onCreate": [
```

Comment:
This empty array can be cleaned up. As above, namespace listeners and model listeners for other parts.

Reply:
removed.